userspace/falco/app/actions/load_config.cpp

```diff
@@ -18,15 +18,67 @@ limitations under the License.
 #include "actions.h"
 #include "falco_utils.h"
 
+/* DEPRECATED: we will remove it in Falco 0.38. */
+#define FALCO_BPF_ENV_VARIABLE "FALCO_BPF_PROBE"
+
 using namespace falco::app;
 using namespace falco::app::actions;
 
-// applies legacy/in-deprecation options to the current config
-static void apply_deprecated_options(
-		const falco::app::options& opts,
-		const std::shared_ptr<falco_configuration>& cfg)
+// applies legacy/in-deprecation options to the current state
+static falco::app::run_result apply_deprecated_options(falco::app::state& s)
 {
-	// Keep for future use cases.
+	// Please note: is not possible to mix command line options and configs to obtain a configuration
+	// we need to use only one method. For example, is not possible to set the gvisor-config through
+	// the command line and the gvisor-root through the config file. For this reason, if we detect
+	// at least one change in the default config we don't allow to use the command line options.
+	if(s.config->m_changes_in_engine_config)
+	{
+		return run_result::ok(); 
```

Comment:
I'm fine with this behavior. However, I don't like silently ignoring deprecated options here (users may not get what's happening).

So, when the user mistakenly uses _BOTH_ the command line flag _AND_ change the config, we should print something like:

_The '-g,--gvisor-config' command line option is deprecated and will be removed in Falco version 0.38. Note that this option is ineffective when 'engine' settings are specified via the configuration file._

Reply:
I'd add a single log, something like:
```
if(s.config->m_changes_in_engine_config)
	{
                   falco_logger::log(falco_logger::level::WARNING, "Since you are using the new engine configuration key, deprecated command line options [-e,-g,...] and FALCO_BPF_PROBE env variable will be ignored.\n");
		return run_result::ok(); 
```

Reply:
Added in latest commit; example output:
```
Thu Nov 23 15:15:50 2023: Falco version: 0.37.0-146+9bf0565 (x86_64)
Thu Nov 23 15:15:50 2023: Falco initialized with configuration file: ../falco.yaml
Thu Nov 23 15:15:50 2023: Since the new 'engine' config key is being used, deprecated CLI options [-e,-g,--gvisor-config,--nodriver,--modern-bpf] and FALCO_BPF_PROBE environment variable will be ignored.
Thu Nov 23 15:15:50 2023: Loading rules from file ../rules/falco_rules.yaml
```

Reply:
Ok. I was thinking of printing the warning only if one of those options was used. But your solution is fine anyway :+1: